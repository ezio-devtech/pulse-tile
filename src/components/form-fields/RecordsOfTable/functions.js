import _ from 'lodash/fp';
import { getDDMMMYYYY } from '../../../utils/time-helpers.utils';

export function changeArraysForTable(arr, name, date) {
  return arr.map((el, index) => {
    el.tableName = el[name];
    el.date = getDDMMMYYYY(el[date]);
    return {
      record: el,
      title: el[name],
      value: index,
    }
  });
}

export function setDiagnosisRecords(data) {
  return changeArraysForTable(data, 'problem', 'dateOfOnset');
}

export function setMedicationsRecords(data) {
  return changeArraysForTable(data, 'name', 'dateCreated');
}
