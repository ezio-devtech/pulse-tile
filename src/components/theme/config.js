import vaccinations from './plugins/Vaccinations/index';

/**
 * The list of none-core plugins which are included to the current theme
 *
 * @return array
 */
export const themePlugins = [

    vaccinations,

];

/**
 * The list of clients URLs for none-core plugins
 *
 * @return array
 */
export const themeClientUrls = {

    VACCINATIONS: 'vaccinations',

};

/**
 * The list of pages for none-core plugins
 *
 * @return array
 */
export const themePluginsPages = {

    'vaccinations': {
      breadcrumbs: [{
        title: 'Vaccinations',
        state: '/vaccinations',
      }],
    },

};

/**
 *
 */
export const themePatientSummaryConfig = [
    {
        key: 'vaccinations',
        title: 'Vaccinations',
        titleCheckboxes: 'Vaccinations',
        state: 'vaccinations',
        nameCheckboxes: 'vaccinations',
        imgPreview: '/images/patients-summary/vaccinations.jpg',
        isDefaultSelected: true,
    },
];
