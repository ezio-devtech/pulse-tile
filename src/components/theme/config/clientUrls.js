/**
 * This information is necessary for plugins URLs
 */
export const themeClientUrls = {
    PATIENTS: '/patients',
};
