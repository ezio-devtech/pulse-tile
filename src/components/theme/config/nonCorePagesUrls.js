/**
 * This is array of NonCore pages
 * (empty for Core)
 */
export const nonCorePagesUrls = [];
