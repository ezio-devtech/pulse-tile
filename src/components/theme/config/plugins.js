/**
 * The list of none-core plugins which are included to the current theme
 *
 * @return array
 */
export const themePlugins = [];

/**
 * This information is necessary for menu and URL
 *
 * @return array
 */
export const themePluginsPages = {};

/**
 * This information is necessary for Patient Summary
 *
 * @return array
 */
export const themePatientSummaryConfig = [];

/**
 * This information is necessary for Unit-tests
 */
export const testStoreContent = {};
