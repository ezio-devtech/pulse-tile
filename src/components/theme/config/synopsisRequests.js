import _ from 'lodash/fp';

const generateFetchListOnMount = (fetchRequest) => {
    return {
        componentDidMount() {
            const { actions, match } = this.props;
            const userId = _.get('params.userId', match);
            if (userId) actions[fetchRequest]({ userId });
        },
        componentWillReceiveProps(nextProps) {
            const { actions, match } = this.props;
            const nextUserId = _.get('match.params.userId', nextProps);
            const userId = _.get('params.userId', match);
            if (nextUserId !== userId) {
                actions[fetchRequest]({ userId: nextUserId });
            }
        },
    }
};

const generateFetchDetailOnMount = (fetchRequest) => {
    return {
        componentDidMount() {
            const { actions, match } = this.props;
            const userId = _.get('params.userId', match);
            const sourceId = _.get('params.sourceId', match);
            if (userId && sourceId) actions[fetchRequest]({ userId, sourceId });
        },
    }
};

const generateArrayOnMount = (requestArray) => {
    return {
        componentDidMount() {
            const { actions, match } = this.props;
            const userId = _.get('params.userId', match);
            requestArray.forEach(item => {
                if (userId) actions[item]({ userId });
            });
        },
        componentWillReceiveProps(nextProps) {
            const { actions, match } = this.props;
            const nextUserId = _.get('match.params.userId', nextProps);
            const userId = _.get('params.userId', match);
            requestArray.forEach(item => {
                if (nextUserId !== userId) {
                    actions[item]({userId: nextUserId});
                }
            });
        },
    }
}

/**
 * This constant returns array of onMount's
 */
export const themeSynopsisOnMount = (generateArrayOnMount([]));

/**
 * This constant returns array of requests
 */
export const themeSynopsisRequests = {};
