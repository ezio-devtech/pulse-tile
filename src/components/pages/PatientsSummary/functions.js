import { get } from 'lodash';
import { themeConfigs } from '../../../themes.config';

/**
 * This function returns initial panels number
 * Default number is 4: Allergies, Medications, Contacts, Problems
 *
 * @return {number}
 */
function getInitialPanelsNumber() {
  const defaultPanelsNumber = 4;
  const hiddenCorePlugins = get(themeConfigs, 'corePluginsToHide', []);
  return defaultPanelsNumber - hiddenCorePlugins.length;
}


/**
 * This function return the number of synopsis panels for none-core plugins
 *
 * @return {number}
 */
function getNonCorePanelsNumber(testStoreContent) {
  let result = 0;
  const plugins = Object.keys(testStoreContent);
  plugins.forEach(item => {
    if (item !== 'fetchFeedsRequest') {
      result++;
    }
  });
  return result;
}

/**
 * This function return the total number of synopsis panels
 *
 * @param testStoreContent
 * @return {number}
 */
export function getPanelsNumber(testStoreContent) {
  const initialNumber = getInitialPanelsNumber();
  const pluginsNumber = getNonCorePanelsNumber(testStoreContent);
  return (pluginsNumber > 0) ? (initialNumber + pluginsNumber) : initialNumber;
}

/**
 * This function counts headings blocks number
 * - it equals 1 by default (SHOW)
 * - if theme setting patientsSummaryHasPreviewSettings is TRUE we have additional block (VIEW OF BOARDS)
 *
 * @return {number}
 */
export function getHeadingsNumber() {
    return themeConfigs.patientsSummaryHasPreviewSettings ? 2 : 1;
}