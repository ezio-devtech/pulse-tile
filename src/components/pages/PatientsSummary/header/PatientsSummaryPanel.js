import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash/fp';
import { get } from 'lodash';
import { lifecycle } from 'recompose';
import { Row } from 'react-bootstrap';
import { Col } from 'react-bootstrap';

import PTCustomInput from './PTCustomInput';
import { unmountOnBlur } from '../../../../utils/HOCs/unmount-on-blur.utils'
import { patientsSummaryConfig } from '../patients-summary.config';
import { themeConfigs } from '../../../../themes.config';
import { dashboardBeing } from '../../../../plugins.config';

import ExtraPatientsSummarySelectors from '../../../theme/components/ExtraPatientsSummarySelectors';

@lifecycle(unmountOnBlur)

export default class PatientsSummaryPanel extends PureComponent {
  static propTypes = {
    onCategorySelected: PropTypes.func.isRequired,
    onViewOfBoardsSelected: PropTypes.func.isRequired,
    selectedCategory: PropTypes.objectOf(PropTypes.bool).isRequired,
    selectedViewOfBoards: PropTypes.objectOf(PropTypes.bool).isRequired,
  };

  state = {
    selected: this.props.selectedCategory,
    selectedViewOptions: this.props.selectedViewOfBoards,
  };

  componentDidUpdate(prevProps, prevState) {
    if (!_.isEqual(prevState.selected, this.state.selected)) this.props.onCategorySelected(this.state.selected);
    if (!_.isEqual(prevState.selectedViewOptions, this.state.selectedViewOptions)) this.props.onViewOfBoardsSelected(this.state.selectedViewOptions);
  }

  toggleCheckbox = key => this.setState((prevState) => {
    const newValue = !_.get(['selected', key])(prevState);
    const keyName = 'isShow_' + key;
    localStorage.setItem(keyName, newValue);
    return _.set(['selected', key], newValue)(prevState);
  });

  toggleRadio = key => this.setState((prevState) => {
    const selectedViewOptions = Object.assign({}, prevState.selectedViewOptions);
    selectedViewOptions.full = false;
    selectedViewOptions.preview = false;
    selectedViewOptions.list = false;
    selectedViewOptions[key] = true;
    return { 'selectedViewOptions': selectedViewOptions };
  });

  render() {
    const { selected, selectedViewOptions } = this.state;
    const { patientsSummaryHasPreviewSettings, boards } = this.props;

    return (
      <div className="dropdown-menu dropdown-menu-panel dropdown-menu-summary">
        <div className="form-group-wrapper">
          <div className="heading">SHOW</div>
          <div className="form-group">
            <Row>
              {patientsSummaryConfig.map((item) => {
                return dashboardBeing[item.key] !== false ?
                  <Col xs={6} sm={4} key={item.nameCheckboxes}>
                    <PTCustomInput
                      type="checkbox"
                      title={item.titleCheckboxes}
                      id={item.nameCheckboxes}
                      name={item.nameCheckboxes}
                      isChecked={selected[item.key]}
                      onChange={this.toggleCheckbox}
                    />
                  </Col> : null
              })}
            </Row>
          </div>

          <ExtraPatientsSummarySelectors boards={boards} toggleCheckbox={this.toggleCheckbox} />

          {(themeConfigs.patientsSummaryHasPreviewSettings || patientsSummaryHasPreviewSettings) ?
            <div>
              <div className="heading">VIEW OF BOARDS</div>
              <div className="form-group">
                <Row>
                  <Col xs={12}>
                    <PTCustomInput type="radio" title="Headings + List" id="full" name="view-of-preview" value="full" isChecked={get(selectedViewOptions, 'full', null)} onChange={this.toggleRadio} />
                  </Col>
                </Row>
                <Row>
                  <Col xs={12}>
                    <PTCustomInput type="radio" title="Headings" id="preview" name="view-of-preview" value="preview" isChecked={get(selectedViewOptions, 'preview', null)} onChange={this.toggleRadio} />
                  </Col>
                </Row>
              </div>
            </div> : null}
        </div>
      </div>
    )
  }
}
