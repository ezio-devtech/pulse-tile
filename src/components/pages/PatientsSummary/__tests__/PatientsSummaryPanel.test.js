import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import PatientsSummaryPanel from '../header/PatientsSummaryPanel';
import { themeConfigs } from '../../../../themes.config';
import { testStoreContent } from '../../../theme/config/plugins';
import { getPanelsNumber, getHeadingsNumber } from '../functions';

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}
global.localStorage = new LocalStorageMock();

class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key] || null;
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    delete this.store[key];
  }
}
global.localStorage = new LocalStorageMock();

const testProps = {
  onCategorySelected: () => {},
  selectedCategory: {
    problems: true,
    contacts: true,
    allergies: true,
    medications: true,
  },
};

configure({ adapter: new Adapter() });

describe('Component <PatientsSummaryPanel />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <PatientsSummaryPanel
        onCategorySelected={testProps.onCategorySelected}
        selectedCategory={testProps.selectedCategory}
        onViewOfBoardsSelected={() => {}}
      />).dive();
    expect(component).toMatchSnapshot();

    expect(component.instance().props.onCategorySelected).toEqual(testProps.onCategorySelected);
    expect(component.instance().props.selectedCategory).toEqual(testProps.selectedCategory);

    const panelsNumber = getPanelsNumber(testStoreContent);
    const headingsNumber = getHeadingsNumber();

    expect(component.find('.heading')).toHaveLength(headingsNumber);
    if (headingsNumber > 1) {
      expect(component.find('.heading').at(0).text()).toEqual('SHOW');
      expect(component.find('.heading').at(1).text()).toEqual('VIEW OF BOARDS');
      expect(component.find('PTCustomInput[type="checkbox"]')).toHaveLength(panelsNumber);
      expect(component.find('PTCustomInput[type="radio"]')).toHaveLength(2);
    } else {
      expect(component.find('.heading').text()).toEqual('SHOW');
      expect(component.find('PTCustomInput[type="checkbox"]')).toHaveLength(panelsNumber);
    }

    component.instance().toggleCheckbox('dashboard-name');
    component.setState({ selected: {
      problems: true,
      contacts: false,
      allergies: true,
      medications: false,
    } });
    component.setState({ selected: {
      problems: true,
      contacts: false,
      allergies: true,
      medications: false,
    } });

    component.instance().toggleRadio('test');

    component.setState({
      selectedViewOptions: {
        full: false,
        preview: true,
      },
    });

    component.setProps({
      patientsSummaryHasPreviewSettings: true,
    })
  });

  it('should renders with all props correctly with LeedsPHRTheme', () => {
    themeConfigs.isLeedsPHRTheme = true;
    const component = shallow(
      <PatientsSummaryPanel
        onCategorySelected={testProps.onCategorySelected}
        selectedCategory={testProps.selectedCategory}
        onViewOfBoardsSelected={() => {}}
        feeds={testProps.feeds}
      />).dive();

    expect(component).toMatchSnapshot();
  });
});
