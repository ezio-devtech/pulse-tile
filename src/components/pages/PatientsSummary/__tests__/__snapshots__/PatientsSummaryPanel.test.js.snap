// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PatientsSummaryPanel /> should renders with all props correctly 1`] = `
<div
  className="dropdown-menu dropdown-menu-panel dropdown-menu-summary"
>
  <div
    className="form-group-wrapper"
  >
    <div
      className="heading"
    >
      SHOW
    </div>
    <div
      className="form-group"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          key="problems"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="problems"
            isChecked={true}
            name="problems"
            onChange={[Function]}
            title="Problems / Diagnosis"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="medications"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="medications"
            isChecked={true}
            name="medications"
            onChange={[Function]}
            title="Medications"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="allergies"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="allergies"
            isChecked={true}
            name="allergies"
            onChange={[Function]}
            title="Allergies"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="contacts"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="contacts"
            isChecked={true}
            name="contacts"
            onChange={[Function]}
            title="Contacts"
            type="checkbox"
          />
        </Col>
      </Row>
    </div>
    <ExtraPatientsSummarySelectors
      toggleCheckbox={[Function]}
    />
  </div>
</div>
`;

exports[`Component <PatientsSummaryPanel /> should renders with all props correctly with LeedsPHRTheme 1`] = `
<div
  className="dropdown-menu dropdown-menu-panel dropdown-menu-summary"
>
  <div
    className="form-group-wrapper"
  >
    <div
      className="heading"
    >
      SHOW
    </div>
    <div
      className="form-group"
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          key="problems"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="problems"
            isChecked={true}
            name="problems"
            onChange={[Function]}
            title="Problems / Diagnosis"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="medications"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="medications"
            isChecked={true}
            name="medications"
            onChange={[Function]}
            title="Medications"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="allergies"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="allergies"
            isChecked={true}
            name="allergies"
            onChange={[Function]}
            title="Allergies"
            type="checkbox"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          key="contacts"
          sm={4}
          xs={6}
        >
          <PTCustomInput
            id="contacts"
            isChecked={true}
            name="contacts"
            onChange={[Function]}
            title="Contacts"
            type="checkbox"
          />
        </Col>
      </Row>
    </div>
    <ExtraPatientsSummarySelectors
      toggleCheckbox={[Function]}
    />
  </div>
</div>
`;
